### PR TITLE
Prevent multiple bot runs in the same test plan report

### PR DESCRIPTION
**Pull request by stalgiag**
refer to title.

There is currently a single avenue by which one can create multiple bot runs in the same test plan report. This is through the Test Queue -> Add Tests to Test Queue flow. This PR prevents this.

### CODE REVIEW

**Comment by jugglinmike** [comment]
Thanks, @stalgiag! This is a little more substantial than I expected, so I'm going to wait to review until I can give it the attention it deserves on Monday.

**Comment by stalgiag** [comment]
I agree. The state needed for the conditional was readily available. That is the source of the majority of the code here but yes, proper review and more deliberate timing is very much welcome!

File: client/components/AddTestToQueueWithConfirmation/index.jsx [hunk 1]
```diff
@@ -23,13 +26,30 @@ function AddTestToQueueWithConfirmation({
...
+
     const { triggerLoad, loadingMessage } = useTriggerLoad();
     const buttonRef = useRef();
 
     const hasAutomationSupport = isSupportedByResponseCollector({
         at,
         browser
     });
+    const alreadyHasBotInTestPlanReport =
+        testPlanReportsInitiatedByAutomation?.testPlanVersion.testPlanReports.some(
```
Comment (stalgiag) [inline client/components/AddTestToQueueWithConfirmation/index.jsx:48]: looking back at this now. I would like to change this to `useMemo`. I will do this later today.

File: client/components/TestQueue/AssignTesterDropdown/index.jsx [hunk 2]
```diff
@@ -132,7 +132,7 @@ const AssignTesterDropdown = ({
                                 ? 'assigned'
                                 : 'not-assigned';
                             let icon;
-                            if (isBot(tester)) {
+                            if (isBot(tester) && !testerIsAssigned) {
```
[the remark this answers is not in the document]
Comment (stalgiag) [reply client/components/TestQueue/AssignTesterDropdown/index.jsx]: Great point! Reads much better now.

**Review by jugglinmike — APPROVED** [review]
Looking good!

**Pull request merged**